crytic_compile/platform/etherscan.py

```diff
@@ -432,13 +443,44 @@ def _convert_version(version: str) -> str:
     return version[1 : version.find("+")]
 
 
-def _relative_to_short(relative: Path) -> Path:
-    """Translate relative path to short (do nothing for etherscan)
+def _sanitize_remappings(
+    remappings: Optional[List[str]], allowed_directory: str
+) -> Optional[List[str]]:
+    """Sanitize a list of remappings
 
     Args:
-        relative (Path): path to the target
+        remappings: (Optional[List[str]]): a list of remappings
+        allowed_directory: the allowed base directory for remaps
 
     Returns:
-        Path: Translated path
+        Optional[List[str]]: a list of sanitized remappings
     """
-    return relative
+
+    if remappings is None:
+        return remappings
+
+    allowed_path = os.path.abspath(allowed_directory)
+
+    remappings_clean: List[str] = []
+    for r in remappings:
+        split = r.split("=", 2)
+        if len(split) != 2:
+            LOGGER.warning("Invalid remapping %s", r)
+            continue
+
+        origin, dest = split[0], PurePosixPath(split[1])
+
+        # if path is absolute, relativize it
+        if dest.is_absolute():
+            dest = PurePosixPath(*dest.parts[1:])
+
+        dest_disk = Path(allowed_directory, dest)
+
+        if os.path.commonpath((allowed_path, os.path.abspath(dest_disk))) != allowed_path:
+            LOGGER.warning("Remapping %s=%s is potentially unsafe, skipping", origin, dest)
+            continue
```

Comment:
Not blocking but we should extract this into utils so we can have a common API to validate absolute path resolution (ref https://github.com/crytic/crytic-compile/commit/8a7c037261ff618964c23cab9f401413bb191783)